### PR TITLE
#4: calling aws rekognition for a new file

### PR DESCRIPTION
**Pull request by sebastianfeduniak**
In this change:
- listening to S3 bucket changes
- labelling added images with AWS Rekognition
- logging a result

### CODE REVIEW

File: cat-rekognition/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "cat-rekognition",
```
Comment (afronski) [inline cat-rekognition/package.json]: I'd vote for common prefix like: `serverless-cat-detector-lambda-rekognition`

File: cat-rekognition/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "1.0.0",
+  "description": "Detecting cat in a picture",
+  "main": "src/handler.js",
+  "dependencies": {
+    "serverless": "^1.24.0"
```
Comment (afronski) [inline cat-rekognition/package.json]: Do not use `^`, pin in it to the exact version.

File: cat-rekognition/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "devDependencies": {
+    "chai": "3.5.0",
+    "mocha": "3.2.0"
+  },
+  "scripts": {
+    "test": "./node_modules/mocha/bin/mocha test"
+  },
+  "author": "sebastian.feduniak@appliscale.io",
+  "license": "ISC"
```
Comment (afronski) [inline cat-rekognition/package.json]: Not true, have a look into _LICENSE.md_.

File: cat-rekognition/serverless.yml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    - Effect: Allow
+      Action:
+        - s3:*
+      Resource: "*"
+    - Effect: "Allow"
+      Action:
+        - "rekognition:*"
+      Resource: "*"
+
+# you can overwrite defaults here
```
Comment (afronski) [inline cat-rekognition/serverless.yml]: Do we need those comments here?

File: cat-rekognition/src/handler.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    const filesToBeChecked = this.recordsToFiles(this.filterEvents(event));
+
+    filesToBeChecked.forEach(function (fileName) {
+        var params = {
+            Image: {
+                S3Object: {
+                    Bucket: "cat-detector-img-repo",
```
Comment (afronski) [inline cat-rekognition/src/handler.js]: As above - let's use common prefix.

File: cat-rekognition/src/handler.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+
+const rekognition = new AWS.Rekognition();
+
+module.exports.imgClassification = (event, context, callback) => {
+
+    console.log('Received event:', JSON.stringify(event, null, 2));
```
Comment (afronski) [inline cat-rekognition/src/handler.js]: `%j` does `JSON.stringify` out of the box.

File: cat-rekognition/.gitignore [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# package directories
+node_modules
+jspm_packages
+
+# Serverless directories
```
Comment (afronski) [inline cat-rekognition/.gitignore]: Fix those comments and make them proper English sentences. :wink:

File: cat-rekognition/src/handler.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        };
+
+        rekognition.detectLabels(params, function(err, data) {
+            if (err) console.log(err, err.stack);
+            else     console.log(data);
+        });
+    });
+
+    callback(null, event);
+
```
Comment (ppikula) [inline cat-rekognition/src/handler.js]: nitpicking: unnecessary empty line?

File: cat-rekognition/src/handler.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+module.exports.filterEvents = (s3Events) => {
+    const records = s3Events['Records'];
+    const fileAddedEvents = records.filter(function (event) {
+        return event['eventName'] == 'ObjectCreated:Put'
+    });
+    return fileAddedEvents;
+};
+
```
Comment (ppikula) [inline cat-rekognition/src/handler.js]: nitpicking: inconsistent number of lines between functions - here we have 2 above only 1

**Pull request merged**